lib/web_ui/lib/src/engine/view_embedder/flutter_view_manager.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine.dart';
+import 'package:ui/src/engine/js_interop/js_app.dart';
+
+/// Encapsulates the `viewData` Map, and automates the creation of `viewId`s.
+class FlutterViewManager {
+  /// Creates a FlutterViewManager with a [initialViewId].
+  ///
+  /// This defaults to [kImplicitViewId].
+  FlutterViewManager({int initialViewId = kImplicitViewId}) : _nextViewId = initialViewId;
+
+  // The counter from which viewIds are generated. Defaults to [kImplicitViewId].
+  int _nextViewId;
+  final Map<int, EngineFlutterView> _viewData = <int, EngineFlutterView>{};
+  final Map<int, JsFlutterViewOptions> _jsViewOptions = <int, JsFlutterViewOptions>{};
+
+  /// Exposes all the [EngineFlutterView]s registered so far.
+  Iterable<EngineFlutterView> get views => _viewData.values;
+
+  /// Retrieves an [EngineFlutterView] by its `viewId`.
+  EngineFlutterView? operator[](int viewId) {
+    return _viewData[viewId];
+  }
+
+  /// Creates (and stores) a view from its [JsFlutterViewOptions].
+  ///
+  /// Returns a `viewId` that is automatically assigned by this method.
+  int createView({ required JsFlutterViewOptions options }) {
```

Comment:
This API feels too much like an SQL database API, with IDs and ID look-up queries. Can this instead return `EngineFlutterView`? This would remove the need to deal with null checks due to nullability of `operator[]`, and if the call site needs the ID, it can call `view.viewId`.

Reply:
Agreed. I would also like for this class to be able to register listeners who want to receive events when views are registered or unregistered

Reply:
> This would remove the need to deal with null checks due to nullability of operator[], and if the call site needs the ID, it can call view.viewId.

Haha yes, this is totally modeled after what I needed in the call site, and not most logical (createView -> return a view)

> register listeners who want to receive events when views are registered or unregistered

Good call, I was wondering how I could hook up the notification of a view being registered/unregistered. I think we can add an `onViewsChanged` broadcast stream or similar, so hooking this up to the framework is as easy as listening to that stream and propagating an event.

@harryterkelsen do you think there's value in having separate events for added/removed views, or is "changed" enough?